src/app/pages/questions/components/question/audio-input/audio-input.component.ts

```diff
@@ -67,59 +69,88 @@ export class AudioInputComponent implements OnChanges, OnDestroy, OnInit {
   ngOnDestroy() {
     this.resumeListener.unsubscribe()
     this.pauseListener.unsubscribe()
-    this.audioRecordService.destroy()
   }
 
-  ngOnChanges() {
-    if (this.currentlyShown) this.startRecording()
+  handleRecording() {
+    if (!this.isRecording()) {
+      this.recordAttempts++
+      if (this.recordAttempts <= DefaultNumberofAudioAttempts) {
+        if (this.recordAttempts > DefaultAudioAttemptThreshold)
+          this.showAttemptAlert()
+        this.transitionButton()
+        this.startRecording().catch(e => this.showTaskInterruptedAlert())
+      }
+    } else {
+      this.stopRecording().catch(e => this.showTaskInterruptedAlert())
+      if (this.recordAttempts == DefaultNumberofAudioAttempts)
+        this.buttonShown = false
+    }
```

Comment:
put this if block as an `else` clause to the `if (this.recordAttempts <= DefaultNumberofAudioAttempts) {` on line 77

Reply:
This is under the `else` of `if(!this.isRecording())` so that after clicking the `stop` button on the last attempt, the button will disappear. If it is under the `if` block the button will disappear after clicking `start`.

Reply:
ah yes sorry my bad.